intents/intents.py

```diff
@@ -0,0 +1,115 @@
+"""This module currently contains the signature/implementation of some intents"""
+
+import aspects
+
+import datetime
+
+def top_k(table, metric, dimensions, is_asc, k, **kwargs):
+    """This function will implement the top-k intent
+
+    Sorts tuples in the order metric, after applying slice, groupby operations. Also removes the tuples that do not lie in the given date range.
+    The arguments 'table, metric,dimension,sort_order, k' are not optional so they are passed as it is,
+    rest of the arguments that are optional('date_range', 'slices') will be passed in kwargs.
+    If the summary_operator is not None, it groups by dimensions.
+
+    Args:
+        table: Type-pandas.dataframe
+            It has the contents of the csv file
+        metric: Type-string
+            It is the name of the column according to which we sort, and in the case when grouping has to be done, summary operator is applied on metric. metric could a column containing strings, if we are applying count operator on it.
+        dimensions: Type-list of str
+            It is the name of column we want. ex query : 'top 5 batsman according to runs' , here dimension is 'batsman', when summary_operator is not None, we group by dimensions.
```

Comment:
How about top runs per batsman? In this case, summary operator is "none", but we still want a grouping and highest run as the output.

Reply:
The query " top runs per batsman? " would be an example of show intent, with grouping by batsman and applying 'max' summary operator to compress multiple rows per batsman.